linode/manager
# fix: [M3-7083 & M3-7175] MarketPlace regex fix + end to end coverage

## Description 📝
This PR achieves a few things:
- fix the overly permissive regex to map stackcripts labels to OCA labels.
- implements an e2e suite for the marketplace, which doubles as validating the fix above (there was no e2e coverage before and this will be quite helpful when migrating those class components into functional ones and avoid state issues such as M3-7045.
- Improve code structure & quality, and utility coverage of the feature
- Remove reliance on hard coded OCAs for the "New apps" section

## Preview 📷
This PR aims to introduce no visual change or regression

## How to test 🧪
1. Pull code locally and make sure no functional or visual regression was introduced by comparing local and production for the OCA create flow (/linodes/create?type=One-Click)
2. Test info drawers for some apps and confirm they are populated
3. run `yarn cy:debug` and run `one-click-apps.spec.ts`
